app/Http/Controllers/Api/ApiV1Controller.php

```diff
@@ -1623,6 +1624,8 @@ public function mediaUpload(Request $request)
         ]);
 
         $user = $request->user();
+        abort_if($user->has_roles && !UserRoleService::can('can-post', $user->id), 403, 'Invalid permissions for this action');
```

Comment:
As every user would have the "default" role, you could remove the `$user->has_roles && ` here and just simplify to `abort_if(!UserRoleService::can('can-post', $user->id), 403, 'Invalid permissions for this action');`

This could also maybe be simplified through the usage of policies and gates: https://laravel.com/docs/10.x/authorization

Reply:
We are using Policies and Gates, UserRoleService is just a cached wrapper around those!